contrib/delta-kernel-rs-cmake/CMakeLists.txt

```diff
@@ -67,6 +67,7 @@ else()
     set(profile "release")
 endif()
 
+file(COPY_FILE ${DELTA_KERNEL_RS_CMAKE_DIR}/Cargo.lock ${DELTA_KERNEL_RS_SOURCE_DIR}/Cargo.lock)
```

Comment:
We can use `FLAGS --lockfile-path` instead

Reply:
Also let's add `--frozen`

Reply:
Also have you figured out why it passed with `--offline` that we have?

Reply:
>We can use FLAGS --lockfile-path instead

Requires nightly, and is unstable.

Reply:
>Also have you figured out why it passed with --offline that we have?

There is a [custom build step](https://github.com/delta-io/delta-kernel-rs/blob/main/ffi/build.rs) which (most likely) didn't respect `--offline` and had to resort to global package registry to collect dependencies. Adding a `Cargo.lock` helped.

Reply:
Don't we already use nightly? (this looks "very useful" for us, since otherwise we are modifying source directory during build, which is always not a good idea)

Reply:
Let's avoid copying files if possible. Also, with a lock file we can enable `LOCKED` in the invocation too.

Finally we need to also update VENDOR.md:


https://github.com/ClickHouse/ClickHouse/blob/249c98791bda80587fe7d0a017f32309d2320507/rust/VENDOR.md?plain=1#L37-L39

Reply:
>Don't we already use nightly?

AFAIK nightly is installed in CI, not sure about developers, they might need to install new toolchains. I'll recheck with nightly and `--lockfile-path`.

Reply:
Or let's just put `Cargo.lock` [here](https://github.com/ClickHouse/delta-kernel-rs)?

Reply:
>not sure about developers, they might need to install new toolchains.

Good point.

>Or let's just put Cargo.lock [here](https://github.com/ClickHouse/delta-kernel-rs)?

I guess it is OK, though it is still unclear how the `--offline` flag allowed it

```
$ git clone https://github.com/delta-io/delta-kernel-rs
$ cargo build --offline
error: no matching package named `object_store` found
location searched: crates.io index
required by package `acceptance v0.8.0 (/Users/azat/src/rust/delta-kernel-rs/acceptance)`
As a reminder, you're using offline mode (--offline) which can sometimes cause surprising resolution failures, if this error is too confusing you may wish to retry without the offline flag.
```

Reply:
It is not `cargo`'s problem, but `cbindgen`, which is called on ARM in the script that I pointed to. AFAIK it didn't reproduce on x86_64 build for me locally.

Reply:
> Or let's just put Cargo.lock [here](https://github.com/ClickHouse/delta-kernel-rs)?

Makes sense.

> AFAIK nightly is installed in CI, not sure about developers, they might need to install new toolchains. I'll recheck with nightly and --lockfile-path.

Nightly is required to build with sanitizers and the vendored modules are tied to the exact version, so it's a project requirement.

Reply:
https://github.com/ClickHouse/delta-kernel-rs/pull/3